codegen/core/src/main/java/software/amazon/smithy/python/codegen/DirectedPythonClientCodegen.java

```diff
@@ -113,31 +102,17 @@ private ProtocolGenerator resolveProtocolGenerator(
 
     @Override
     public void customizeBeforeShapeGeneration(CustomizeDirective<GenerationContext, PythonSettings> directive) {
```

Comment:
SJ has schemas generated in the `customizeBeforeIntegrations`, but I guess there's no practical difference with type codegen? Just something to point out.